packages/api-plugin-fulfillment/src/queries/fulfillmentMethodsWithQuotes.js

```diff
@@ -0,0 +1,51 @@
+import Logger from "@reactioncommerce/logger";
+import ReactionError from "@reactioncommerce/reaction-error";
+import extendCommonOrder from "../util/extendCommonOrder.js";
+
+const logCtx = { name: "fulfillment", file: "fulfillmentMethodsWithQuotes" };
+
+/**
+ * @name fulfillmentMethodsWithQuotes
+ * @method
+ * @summary Just gets rates, without updating anything
+ * @param {Object} commonOrder - details about the purchase a user wants to make.
+ * @param {Object} context - Context
+ * @returns {Array} return updated rates in cart
+ * @private
+ */
+export default async function fulfillmentMethodsWithQuotes(commonOrder, context) {
+  const rates = [];
+  const retrialTargets = [];
+
+  // must have items to calculate shipping
+  if (!commonOrder.items || !commonOrder.items.length) {
+    Logger.debug(logCtx, "fulfillmentMethodsWithQuotes called with CommonOrder with no items");
+    return rates;
+  }
+  const commonOrderExtended = await extendCommonOrder(context, commonOrder);
+
+  const fulfillmentTypeInGroup = commonOrder.fulfillmentType;
+  if (!fulfillmentTypeInGroup) throw new ReactionError("not-found", "Fulfillment type not found in commonOrder");
+
+  const allFuncsArray = context.getFunctionsOfType("fulfillmentMethodsWithQuotes");
+  const ffTypeFuncObjects = allFuncsArray.filter((func) => fulfillmentTypeInGroup === func.key);
+  const funcs = ffTypeFuncObjects.map((func) => func.handler);
+
+  if (!funcs || !Array.isArray(funcs) || !funcs.length) throw new ReactionError("not-found", `No methods for Fulfillment type ${fulfillmentTypeInGroup}`);
```

Comment:
Using lodash can shorten this function

```suggestion
  if (_.isEmpty(funcs)) throw new ReactionError("not-found", `No methods for Fulfillment type ${fulfillmentTypeInGroup}`);
```

Reply:
Thank you. Updated. Will be pushed soon